crates/notion-core/src/session.rs

```diff
@@ -92,4 +126,30 @@ impl Session {
         let config = self.config.get()?;
         catalog.activate_node(matching, config)
     }
+
+    pub fn add_event_start(&mut self, activity_kind: ActivityKind) {
+        self.event_log.add_event_start(activity_kind)
+    }
+    pub fn add_event_end(&mut self, activity_kind: ActivityKind, exit_code: Option<i32>) {
+        self.event_log.add_event_end(activity_kind, exit_code)
+    }
+    pub fn add_event_error(&mut self, activity_kind: ActivityKind, error: &NotionError) {
+        self.event_log.add_event_error(activity_kind, error)
+    }
+
+    // send the events from this session to the monitor
+    pub fn send_events(&mut self) {
+        let command = self.events_command();
+        self.event_log.send_events(command)
+    }
+
+    pub fn events_command(&self) -> Option<String> {
```

Comment:
I prefer to avoid `match` expressions for types like Option. I like `.map`/`.and_then` chaining, but it's a little subtle with `.as_ref()`:
```rust
self.project
    .as_ref()
    .and_then(|project| project.manifest().events_plugin.as_ref())
    .map(|plugin| plugin.to_string())
```
Or you can do two nested `if let`s:
```rust
if let Some(ref project) = self.project {
    if let Some(ref p) = project.manifest().events_plugin {
        return Some(p.to_string());
    }
}
None
```